scripts/eurostat/birth_death_migration/import_data.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas as pd
+
+
+def download_data():
+    """Downloads raw data from Eurostat website and stores it in instance
+    data frame.
+    """
+    raw_df = pd.read_table(DATA_LINK)
+    return raw_df
+
+
+def preprocess_data(raw_df):
+    """Preprocesses instance raw_df and puts it into long format."""
+    if raw_df is None:
+        raise ValueError("Uninitialized value of raw data frame. Please "
+                         "check you are calling download_data before "
+                         "preprocess_data.")
+    assert len(raw_df.columns) > 1, "Data must have at least two columns."
+    assert raw_df.columns.values[0].endswith(
+        '\\time'), "Expected the first column header to end with '\\time'."
+
+    # Convert from one-year per column (wide format) to one row per
+    # data point (long format).
+    preprocessed_df = raw_df.melt(id_vars=[raw_df.columns[0]])
+    # Rename the variable column.
+    preprocessed_df.columns.values[1] = 'time'
+
+    # Split the index column into two.
+    # Ex: 'unit,sex,age,geo\time' -> 'unit,sex,age' and 'geo'.
+    # '\time' labels the other columns so it is confusing.
+    # Replace value column with extra space.
+    preprocessed_df.value = preprocessed_df.value.str.replace(
+        "([0-9:])$", lambda m: m.group(0) + ' ')
+    statistical_variable = None
+    # In case this returns only one element in the list.
+    first_column_list = preprocessed_df.columns[0].rsplit(sep=",", maxsplit=1)
+    if len(first_column_list) == 2:
+        statistical_variable, geo = first_column_list
+    else:
+        geo = first_column_list[0]
+    geo = geo.replace(r'\time', '')
+    assert geo == "geo", "Expected the column header to end with 'geo'."
+
+    if statistical_variable:
+        split_df = preprocessed_df[preprocessed_df.columns[0]].str.rsplit(
```

Comment:
For readability and safety, please use column names instead of indices whenever possible

Reply:
Agreed. It's just this preprocess_data function is kept a bit general so that even if they change the column names, as long as the order is maintained, the script can still run. That is why I didn't hard code the column name here (which should be 'indic_de,geo\time')

Reply:
We would rather fail than have the wrong value, which is why we prefer names. Does that make sense?